Makefile

```diff
@@ -0,0 +1,46 @@
+all: build fmt imports vet lint test
```

Comment:
I'm realizing that we never documented our build script in the ./hack/README.md.  `./hack/go.sh` handles most of these operations, excluding 'vendor'.  

The script handles some known issues with testing (see below), runs a preconfigured `go imports` so we always structure imports uniformly between devs, executes robust linting, and wraps up a combination of these for CI to run.  This also means devs can run CI workflow locally to get a quick glance on whether it will pass.  Obviously this last part isn't a perfect recreation of CI as env's vary, but it's is still useful.